app/mender.go

```diff
@@ -555,13 +571,53 @@ func spinEventLoop(c *ControlMapPool, to State, ctx *StateContext, controller Co
 	}
 }
 
+func refetch(controller Controller) {
+	attempts := 0
+	for {
+		wait, err := client.GetExponentialBackoffTime(
+			attempts,
+			controller.GetUpdatePollInterval())
+		log.Infof("Next Update Check attempt in %d seconds", wait)
+		if err == client.MaxRetriesExceededError {
+			log.Errorf("Failed to refetch the update control map in %d attempts, giving up", attempts)
+			return
+		} else if err != nil {
+			log.Errorf("Unexpected error: %s", err.Error())
+			return
+		}
+		time.Sleep(wait)
+		_, err = controller.CheckUpdate()
+		attempts += 1
```

Comment:
Simply started the count at 1 instead

Reply:
I was about to write that then the log statement below would be wrong then, before realizing that I'm actually wrong! The original is correct. In my (very contrived) example above, the `MaxRetriesExceededError` will only happen on the second iteration, if there is one retry available, in which case it will write `1 attempts`, IOW correct. Sorry for the confusion.